### PR TITLE
georef: fix ambiguity on nearest_addr when we have equal ways

### PR DESCRIPTION
**Pull request by TeXitoi**
http://jira.canaltp.fr/browse/NAVITIAII-2001

### CODE REVIEW

**Comment by TeXitoi** [comment]
Waiting for artemis

File: source/type/geographical_coord.h [hunk 1]
```diff
@@ -65,8 +65,7 @@ struct GeographicalCoord{
     constexpr static double coord_epsilon = 1e-15;
     /// Ordre des coordonnées utilisé par ProximityList
     bool operator<(const GeographicalCoord &other) const {
-        if ( fabs(lon() - other.lon()) > coord_epsilon )
-            return lon() < other.lon();
+        if (lon() != other.lon()) { return lon() < other.lon(); }
```
Comment (pbougue) [inline source/type/geographical_coord.h]: floating != comparison will raise warnings

File: source/georef/georef.cpp [hunk 2]
```diff
@@ -670,10 +647,10 @@ std::pair<int, const Way*> GeoRef::nearest_addr(const type::GeographicalCoord& c
 
     // then, we search in each way the nearest number or way centroid
     std::pair<int, const Way*> result = {0, way_dist.begin()->first};
-    float min_dist = way_dist.begin()->second;
+    double min_dist = way_dist.begin()->second;
     for (const auto& w_d: way_dist) {
         // way centroid
-        if (w_d.second < min_dist) {
+        if (w_d.second < min_dist || (w_d.second == min_dist && w_d.first->uri < result.second->uri)) {
```
Comment (pbougue) [inline source/georef/georef.cpp:656]: floating == comparison will raise warnings

File: source/georef/georef.cpp [hunk 3]
```diff
@@ -681,7 +658,7 @@ std::pair<int, const Way*> GeoRef::nearest_addr(const type::GeographicalCoord& c
         // number
         const auto &nb_dist = w_d.first->nearest_number(coord);
         if (nb_dist.first <= 0) { continue; }
-        if (nb_dist.second <= min_dist) {
+        if (nb_dist.second < min_dist || (nb_dist.second == min_dist && w_d.first->uri < result.second->uri)) {
```
Comment (pbougue) [inline source/georef/georef.cpp:664]: floating == comparison will raise warnings

Comment (pbougue) [inline source/georef/georef.cpp:664]: Done

**Comment by antoine-de** [comment]
OK for me

**Pull request merged**